### DISCUSSION TITLE
Make switchover via operator

### DISCUSSION THREAD

**Discussion by FactorT** [zalando/postgres-operator#1915] [General]
Hello everyone!
I’m investigating “Postgres Operator” and I’ve got a question.
Could you tell me how I can run a swithcover via operator, please?
I know that I can enter in pod and run switchover via patronictl.
But in my humble opinion it’s not optimal way for this operation and do it via operator is smarter.
But can operator do it?

**Comment by apa64** [comment]
We're seeing this as an issue in our clusters. A preStopHook should start switchover during gracefulShutdown period.

The problem is this: Currently the operator seems to set up PDBs which really do not help. The PDB that targets the master node is set at minAvailable 1 but there's only 1 master anyway. During cluster administration (eg. when a node is drained) this keeps the pod running. Now the cluster admin sees that there are some pods which are not moving by themselves to other nodes and does a force delete for those. Administration tasks can proceed but the PostgreSQL master is killed abrubtly. Yes, perhaps the cluster admin should call the db admins but you know how these things sometimes are.  

It would be great if the operator would set preStopHook for the master pod which would start the switchover when a pod is going to get killed. Actually that should remove the need for the PDB because the master node pod could handle its state automatically and it would be just part of the "cattle" .

There's also a warning-level alert up all the time because the pod count is at PDB minimum limit and let's not even get into https://github.com/zalando/postgres-operator/issues/3020.
